rbtcwrapperproxy/contracts/RBTCWrapperProxy.sol

```diff
@@ -446,55 +445,107 @@ contract RBTCWrapperProxy is ContractRegistryClient {
 
             IWrbtcERC20(wrbtcTokenAddress).withdraw(_targetTokenAmount);
 
-            (bool successOfSendRBTC,) = msg.sender.call.value(_targetTokenAmount)("");
+            (bool successOfSendRBTC,) = recipient.call.value(_targetTokenAmount)("");
             require(successOfSendRBTC, "Failed to send RBTC to user");
 
-            emit TokenConverted(msg.sender, _amount, _targetTokenAmount, _path);
+            emit TokenConverted(recipient, _amount, _targetTokenAmount, _path);
 
             return _targetTokenAmount;
         }        
     }
 
     /**
-     * @notice provides funds to a lending pool and deposits the pool tokens into the liquidity mining contract.
-     * @param loanTokenAddress the address of the loan token (aka lending pool)
-     * @param depositAmount he amount of underlying tokens to deposit 
+     * @dev
+     * The conversion function to do the refund.
+     * This function will be called in order to refund the excess of non-wrbtc token in form of RBTC.
+     *
+     * @param _sourceToken the non-wrbtc token which will be converted to WRBTC, and then sent the RBTC to the sender.
+     * @param _amount amount of _sourceToken for refund.
+     *
+     * @return THe total wrbtc refunded.
      */
-    function addToLendingPool(address loanTokenAddress, uint256 depositAmount) public{
-        LoanToken loanToken = LoanToken(loanTokenAddress);
-        IERC20Token underlyingAsset = IERC20Token(loanToken.loanTokenAddress());
+    function convertForRBTCRefund(
```

Comment:
again, i do no not see the need for this.

Reply:
Removed